spec/factories.rb

```diff
@@ -1,5 +1,20 @@
 FactoryGirl.define do
   factory :redirection do
+    sequence(:slug) { |n| "slug#{n}" }
+    sequence(:url) { |n| "http://example#{n}.com" }
+    next_id 0
+
+    after(:create) do |redirection|
+      if redirection.next_id != 0
```

Comment:
This is a huge hack: if the ID is 0 (and thus obviously fake), set `next_id` to refer to a real Redirection.